caddyhttp/fastcgi/fastcgi.go

```diff
@@ -100,6 +102,20 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error)
 			case "OPTIONS":
 				resp, err = fcgiBackend.Options(env)
 			default:
+				// since some chrome releases or quic clients are not sending the Content-Length
+				// in the request headers, caddy should re-calculate content-length by the body and fix it.
+				if contentLength == 0 {
+					var body []byte
+					if r.Body != nil {
+						body, _ = ioutil.ReadAll(r.Body)
```

Comment:
Unfortunately, this line is a show-stopper right now. Not only are we ignoring the error, but we're reading the whole stream into memory over the network. This presents a number of problems, as you can imagine.

I think the whole reason the client is required by spec to send a Content-Length header is so that the server doesn't have to buffer the whole thing first; if the client isn't sending that, doesn't it mean the client is broken, and not Caddy?